<launch>
  <include file="$(find octomap_server)/launch/octomap_mapping.launch"/>
  <node pkg="bezier_exploration" type="bezier_exploration_test" name="bezier_exploration_test" output="screen" clear_params="true">
  </node>

  <node pkg="bezier_exploration" type="bezier_exploration_node" name="bezier_exploration_node" output="screen" clear_params="true">
    <param name="rrt/safe_distance" value="0.3"/>
    <param name="rrt/max_radius" value="3.0"/>
    <param name="rrt/min_radius" value="0.5"/>
    <param name="rrt/tree_max_dim" value="20"/>
    <param name="rrt/max_iterations" value="20"/>
    <param name="rrt/minimum_delta" value="0.5"/>
    <param name="rrt/maximum_delta" value="5.0"/>
    <param name="rrt/time_resolution" value="0.5"/>
    <param name="rrt/weight_time" value="0.5"/>
    <param name="rrt/weight_acc" value="0.1"/>
    <param name="rrt/weight_yaw" value="0.1"/>
    <param name="rrt/max_traj_number" value="3"/>
    <param name="rrt/verbose" value="true"/>
    <param name="rrt/verbose_debug" value="false"/>
    <param name="rrt/verbose_vis" value="true"/>
    <param name="uav/max_velocity" value="1.5"/>
    <param name="uav/max_acceleration" value="2.0"/>
    <param name="map/x_max" value="18.5"/>
    <param name="map/x_min" value="-0.5"/>
    <param name="map/y_max" value="4.5"/>
    <param name="map/y_min" value="-4.5"/>
    <param name="map/z_max" value="3.0"/>
    <param name="map/z_min" value="0.5"/>
  </node>

  <node pkg="bezier_exploration" type="gain_evaluator_node" name="gain_evaluator_node" output="screen" clear_params="true">
    <param name="rrt/horizontal_fov" value="115"/>
    <param name="rrt/vertical_fov" value="60"/>
    <param name="rrt/delta_r" value="0.1"/>
    <param name="rrt/min_r" value="0.3"/>
    <param name="rrt/max_r" value="5.0"/>
    <param name="rrt/delta_phi" value="10.0"/>
    <param name="rrt/delta_theta" value="10"/>

    <param name="map/x_max" value="18.5"/>
    <param name="map/x_min" value="-0.5"/>
    <param name="map/y_max" value="4.5"/>
    <param name="map/y_min" value="-4.5"/>
    <param name="map/z_max" value="2.5"/>
    <param name="map/z_min" value="0.5"/>
  </node>
<!--
  <node pkg="bezier_exploration" type="gain_regressor.py" name="gain_regressor" output="screen" clear_params="true">
    <param name="gp/l_x" value="2.0"/>
    <param name="gp/l_y" value="2.0"/>
    <param name="gp/l_z" value="2.0"/>
    <param name="gp/sigma_n" value="0.001"/>
    <param name="gain/insertion_range" value="1.0"/>
    <param name="gain/gp_evaluation_range" value="5.0"/>
  </node>
  -->
  
  <node pkg="bezier_exploration" type="map_evaluator" name="map_evaluator" output="screen" clear_params="true">
  </node>
</launch>
